src/renderer/screens/exchange/Buy/SelectAccountAndCurrency.js

```diff
@@ -96,6 +96,65 @@ const SelectAccountAndCurrency = ({ selectAccount, defaultCurrency, defaultAccou
     dispatch(openModal("MODAL_ADD_ACCOUNTS", { currency }));
   }, [dispatch, currency]);
 
+  const addOrSelectAccount = useMemo(() => {
+    if (currency) {
+      return availableAccounts.length ? (
+        <>
+          <FormContent>
+            <AccountSelectorLabel>
+              <span>{t("exchange.buy.selectAccount")}</span>
+              <FakeLink fontSize={3} ff="Inter|SemiBold" onClick={openAddAccounts}>
+                <PlusIcon size={10} />
+                <Text style={{ marginLeft: 4 }}>{t("exchange.buy.addAccount")}</Text>
+              </FakeLink>
+            </AccountSelectorLabel>
+            <SelectAccount
+              accounts={availableAccounts}
+              value={{ account, subAccount }}
+              onChange={setAccount}
+            />
+          </FormContent>
+          <FormContent>
+            <ConfirmButton
+              primary
+              onClick={() => {
+                if (account) {
+                  track("Buy Crypto Continue Button", {
+                    currencyName: getAccountCurrency(account).name,
+                    isEmpty: isAccountEmpty(account),
+                  });
+                  if (subAccount) {
+                    selectAccount(subAccount, account);
+                  } else {
+                    selectAccount(account);
+                  }
+                }
+              }}
+              disabled={!account}
+            >
+              {t("exchange.buy.continue")}
+            </ConfirmButton>
+          </FormContent>
+        </>
+      ) : (
+        <FormContent>
+          <ConfirmButton primary onClick={openAddAccounts}>
+            {t("exchange.buy.addAccount")}
+          </ConfirmButton>
+        </FormContent>
+      );
+    }
```

Comment:
As a rule of thumb and in order to improve readability, it's a good practice to favor early returns

```suggestion
    if (!currency) {
      return;
    }

    if (!availableAccounts.length) {
      return (
        <FormContent>
          <ConfirmButton primary onClick={openAddAccounts}>
            {t("exchange.buy.addAccount")}
          </ConfirmButton>
        </FormContent>
      );
    }

    return (
      <>
        <FormContent>
          <AccountSelectorLabel>
            <span>{t("exchange.buy.selectAccount")}</span>
            <FakeLink fontSize={3} ff="Inter|SemiBold" onClick={openAddAccounts}>
              <PlusIcon size={10} />
              <Text style={{ marginLeft: 4 }}>{t("exchange.buy.addAccount")}</Text>
            </FakeLink>
          </AccountSelectorLabel>
          <SelectAccount
            accounts={availableAccounts}
            value={{ account, subAccount }}
            onChange={setAccount}
          />
        </FormContent>
        <FormContent>
          <ConfirmButton
            primary
            onClick={() => {
              if (!account) {
                return;
              }

              track("Buy Crypto Continue Button", {
                currencyName: getAccountCurrency(account).name,
                isEmpty: isAccountEmpty(account),
              });

              subAccount ? selectAccount(subAccount, account) : selectAccount(account);
            }}
            disabled={!account}
          >
            {t("exchange.buy.continue")}
          </ConfirmButton>
        </FormContent>
      </>
    );
```

Reply:
👍